src/bootstrap/src/core/builder/cargo.rs

```diff
@@ -779,6 +779,12 @@ impl Builder<'_> {
             Mode::Codegen => metadata.push_str("codegen"),
             _ => {}
         }
+        // `rustc_driver`'s version number is always `0.0.0`, which can cause problems on
+        // side-by-side installs because we don't include the version number of the `rustc_driver`
+        // being built. This can cause non-stable channel builds producing artifacts that end up
+        // with identical `librustc_driver*.so` filename hashes.
```

Comment:
FWIW I'm not *super* happy with this comment, lmw if there's a better wording for it.

Reply:
this is a problem regardless of the channel. it just happens to not trigger on stable because cargo uses different heuristics per-channel for what to include in the metadata hash.